crates/rpc/rpc/src/admin.rs

```diff
@@ -13,13 +14,13 @@ pub struct AdminApi<N> {
     /// An interface to interact with the network
     network: N,
     /// The specification of the blockchain's configuration.
-    chain_spec: ChainSpec,
+    chain_spec: Arc<ChainSpec>,
 }
 
 impl<N> AdminApi<N> {
     /// Creates a new instance of `AdminApi`.
     pub fn new(network: N, chain_spec: ChainSpec) -> Self {
```

Comment:
here as well because the chainspec provider also returns an Arc of this, so we don't need to duplicate it
```suggestion
    pub fn new(network: N, chain_spec: Arc<ChainSpec>) -> Self {
```

Reply:
My bad you are right, didn't notice that, just fixed